### PR TITLE
fix: use proxy when copying node with successors

### LINKED ISSUE #383: node not found when using oras.Copy with certain option
When using `oras.Copy` to copy the below DAG, if `FindSuccessors` is provided and successors of `a` is obtained without fetching , then a not found error will be thrown when copying node `a`.
```
   +-> b
a -+-> c
   +-> d
```

### PR DESCRIPTION
**Pull request by qweeah**
Resolves #383 
Signed-off-by: Billy Zha <jinzha1@microsoft.com>

### CODE REVIEW

File: copy.go [hunk 1]
```diff
@@ -250,7 +250,7 @@ func copyGraph(ctx context.Context, src content.ReadOnlyStorage, dst content.Sto
 		if err := region.Start(); err != nil {
 			return err
 		}
-		return copyNode(ctx, proxy.Cache, dst, desc, opts)
+		return copyNode(ctx, proxy, dst, desc, opts)
```
Comment (shizhMSFT) [inline copy.go:253]: If the content is not in the proxy cache, shouldn't copy from the source directly just like the copy operations for leaf node?

Comment (qweeah) [reply copy.go:253]: Copying from the source is the safiest choice, but in that case `proxy` won't be used at all, is what we expected?

Comment (qweeah) [reply copy.go]: A manifest needs not to be cached if its successors are externally provided.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**